tools/coqdep_common.ml

```diff
@@ -127,60 +129,74 @@ let get_prefix p l =
   in
   drop_prefix_rec (p, l)
 
-let search_table (type r) is_root table ?from s = match from with
-| None -> Hashtbl.find table s
+let search_table is_root get table clash ?from s = match from with
+| None -> (try StrListMap.find s clash with Not_found -> [get (Hashtbl.find table s)])
 | Some from ->
-  let module M = struct exception Found of r end in
-  let iter logpath binding =
+  let fold logpath binding acc =
     if is_root binding then match get_prefix from logpath with
-    | None -> ()
+    | None -> acc
     | Some rem ->
       match get_prefix (List.rev s) (List.rev rem) with
-      | None -> ()
-      | Some _ -> raise (M.Found binding)
+      | None -> acc
+      | Some _ -> get binding :: acc
+    else
+      acc
   in
-  try Hashtbl.iter iter table; raise Not_found
-  with M.Found s -> s
+  (* We reverse to give priority for compatibility. There is an issue
```

Comment:
Can you expand on the compatibility requirements. Is anyone relying on this behaviour?

Reply:
hierarchy-builder was relying on this order.